source/features/warn-pr-from-master.tsx

```diff
@@ -1,15 +1,16 @@
 import React from 'dom-chef';
-import {$, elementExists} from 'select-dom';
+import {$} from 'select-dom';
 import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager.js';
 import getDefaultBranch, {defaultBranchOfRepo} from '../github-helpers/get-default-branch.js';
 import {getRepo} from '../github-helpers/index.js';
+import observe from '../helpers/selector-observer.js';
 
-async function init(): Promise<false | void> {
+async function warnPR(crossRepo: Element): Promise<false | void> {
 	let defaultBranch;
-	if (elementExists('.is-cross-repo')) {
-		const forkedRepository = getRepo($('[title^="head: "]')!.textContent)!;
+	if (crossRepo) {
+		const forkedRepository = getRepo($('.range-cross-repo-pair:has(#head-ref-selector) span[data-menu-button]')!.textContent)!;
```

Comment:
This needs further changes actually. This condition is now always true, so `else` is never called.

But I don’t think it ever made sense, it was added in https://github.com/refined-github/refined-github/pull/3511

So:

- drop the `if(crossRepo)` line
- drop the entire `else` part
- Move the `data-menu-button` selector to the observer

Reply:
I'm surprised that neither TS nor ESLint is complaining that the condition is always truthy 😬

Reply:
This is weird actually, we don't need to read this data from the DOM, it's already in the URL, it's being checked just later.

So probably:

- keep `observe('.is-cross-repo'`
- read `forkedRepository` from the `...user:repo` URL part instead


Also make sure that renamed repos are supported, test URL:

- https://github.com/refined-github/sandbox/compare/default-a...bfred-it-org:github-sandbox:main?expand=1